src/network/core.rs

```diff
@@ -232,6 +232,180 @@ impl Core {
         }
     }
 
+    pub fn macvlan_per_podman_network(
+        per_network_opts: &types::PerNetworkOptions,
+        network: &types::Network,
+        netns: &str,
+    ) -> Result<types::StatusBlock, std::io::Error> {
+        //  StatusBlock response
+        //  StatusBlock response
+        let mut response = types::StatusBlock {
+            interfaces: Some(HashMap::new()),
+        };
+        // Does config have a macvlan mode ? I think not
+        // Important !! Hardcode MACVLAN_MODE to bridge
+        let macvlan_mode: u32 = 4u32;
```

Comment:
We support setting macvlan modes, see https://github.com/containers/podman/blob/main/docs/source/markdown/podman-network-create.1.md#--optoption--o

You have to read the `mode` field from the network options and use the correct mode. If it is unset default to bridge.

Reply:
ah sure i'll add support for this probably `enum` map was not aware that we were supporting all types.

Reply:
@Luap99 added support for all available `macvlan` modes in latest commit.